### PR TITLE
Fixed seek offset size to 64bit.

### LINKED ISSUE #26998: torch.load() - OSError on a file larger than 2GB
## 🐛 Bug
In version 1.2, torch.load() does not work with files larger than 2GB.
```python
import torch
import pathlib
torch.save(torch.rand(int((2 * 1024**3) / 4), dtype=torch.float32, device="cpu"),
           pathlib.Path("pytorch_test_load.temp"))
torch.load(pathlib.Path("pytorch_test_load.temp"))
```
```
 File "torch\serialization.py", line 581, in _load
    deserialized_objects[key]._set_from_file(f, offset, f_should_read_directly)
OSError: [Errno 22] Invalid argument
```
This is probably due to https://github.com/pytorch/pytorch/pull/20270 or https://github.com/pytorch/pytorch/pull/20900

Windows 10
Python 3.7
Pytorch 1.2.0

### PR DESCRIPTION
**Pull request by yn4bit**
Fixes #26998

### CODE REVIEW

**Review by peterjc123 — APPROVED** [review]

**Comment by peterjc123** [comment]
Could you please add a test as well?

**Comment by yn4bit** [comment]
@peterjc123
I tried to modify the existing test to check read / write of files over 2GB. Any concerns with this approach?

**Comment by peterjc123** [comment]
@yn4bit That's great! Thank you.

**Review by ezyang — APPROVED** [review]

**Comment by ezyang** [comment]
It's a little bad since Windows tests are failing on master but I'm going to optimistically go ahead and land this.

**Comment by ezyang** [comment]
There's something wrong with the tests:

```

16:36:07 ======================================================================
16:36:07 ERROR: test_serialization_offset (__main__.TestTorch)
16:36:07 ----------------------------------------------------------------------
16:36:07 Traceback (most recent call last):
16:36:07   File "test_torch.py", line 5034, in test_serialization_offset
16:36:07     b_loaded = torch.load(f)
16:36:07   File "C:\Jenkins\workspace\pytorch-builds\pytorch-win-ws2016-cuda9-cudnn7-py3-test2\build\win_tmp\build\torch\serialization.py", line 426, in load
16:36:07     return _load(f, map_location, pickle_module, **pickle_load_args)
16:36:07   File "C:\Jenkins\workspace\pytorch-builds\pytorch-win-ws2016-cuda9-cudnn7-py3-test2\build\win_tmp\build\torch\serialization.py", line 620, in _load
16:36:07     deserialized_objects[key]._set_from_file(f, offset, f_should_read_directly)
16:36:07   File "C:\Jenkins\Miniconda3\lib\tempfile.py", line 485, in func_wrapper
16:36:07     return func(*args, **kwargs)
16:36:07 OSError: [Errno 22] Invalid argument
16:36:07 
```

**Comment by ezyang** [comment]
I'm going to revert this for now until we figure out what exactly the problem is

**Comment by peterjc123** [comment]
@yn4bit This one gets reverted, could you please figure out a new one?

File: torch/csrc/generic/StorageMethods.cpp [hunk 1]
```diff
@@ -267,7 +267,7 @@ static PyObject *THPStorage_(setFromFile)(THPStorage *self, PyObject *args)
   // advanced postion
   const auto fd_current_pos = lseek(fd, 0, SEEK_CUR);
```
Comment (peterjc123) [inline torch/csrc/generic/StorageMethods.cpp]: After reading the MSVC document [here](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/lseek-lseeki64?view=vs-2019), looks like we need to change the function too.

**Comment by yn4bit** [comment]
@ peterjc123 Thank you for your comment!

I think lseek is certainly suspicious. Python portable_lseek () is as follows, so I think that it needs to be changed.

#ifdef MS_WINDOWS
     res = _lseeki64(fd, pos, whence);
#else
     res = lseek(fd, pos, whence);
#endif

However, since OSError has finally occurred, I think that the cause of the direct error is the value passed to the Python seek or a restriction inside Python. I would like to correct this after analyzing the generation mechanism, but there is no Windows development environment at hand ...

**Comment by peterjc123** [comment]
@yn4bit I'm testing the fix at #27215. You could comment there if you have more thoughts.

**Comment by yn4bit** [comment]
@peterjc123 Thank you for your testing. Wow, this is a surprise to solve the problem. I don't understand the behavior of Windows ...
I checked the patch and it's looks good to me.

**Pull request closed without merging**